arm-servicebus/2014-09-01/swagger/servicebus.json

```diff
@@ -566,153 +572,2046 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/authorizationRules/{authorizationRuleName}/regenerateKeys": {
+      "post": {
+        "tags": [
+          "Topics"
+        ],
+        "operationId": "Topics_RegenerateKeys",
+        "description": "Primary and Secondary ConnectionStrings to the topic ",
+        "externalDocs": {
+          "url": "https://msdn.microsoft.com/en-us/library/azure/mt720679.aspx"
+        },
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "namespaceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The namespace name."
+          },
+          {
+            "name": "topicName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The topic name."
+          },
+          {
+            "name": "authorizationRuleName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The connection string of the topic for the specified authorizationRule."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RegenerateKeysParameters"
+            },
+            "description": "Parameters supplied to regenerate Auth Rule."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful",
+            "schema": {
+              "$ref": "#/definitions/ResourceListKeys"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics": {
+      "get": {
+        "tags": [
+          "Topics"
+        ],
+        "operationId": "Topics_GetAll",
+        "description": "Enumerates the topics in a namespace. An empty feed is returned if no topic exists in the namespace.",
+        "externalDocs": {
+          "url": "https://msdn.microsoft.com/en-us/library/azure/mt639388.aspx"
+        },
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group. If resourceGroupName value is null the method lists all the namespaces within subscription"
```

Comment:
/cc @tbombach 

> Oh wow!! So if the resourceGroupName is null then the service provides a list of all the namespaces in the subscription.

As of Swagger 2.0 all the path parameters are required which means they cannot be null. If null the generated code will throw a validation exception.

To solve this issue, please add another path
`/subscriptions/{subscriptionId}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics`

OR 

`/subscriptions/{subscriptionId}/resourceGroups/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics`

whichever seems fit for listing resources in a subscription.

Please name that method "Topics_List".

This is a general comment for all the List methods that fall into this category.